### DISCUSSION TITLE
is nextflow mixing up channel data when run in parallel?

### DISCUSSION THREAD

**Discussion by DanielAmsel** [nextflow-io/nextflow#1954] [Q&A] [answered]
## Bug report 

Hi, 
I wrote a nextflow script that takes input data and does something with the data in several dockerized processes.
In a final step, I want to merge the results of different processes into one process that creates a final output.
Nextflow runs in parallel for several data samples, each with a different ID.

### Expected behavior and actual behavior

The final merged results should come from data with the same ID, since it was the only input for the individual nextflow run.

Actually it seems that nextflow mixes up data from other runs that are performed in parallel.
In the work directory, I also find folders with different IDs in it.

Example:
Input data ID-1 --> processing --> result comes from ID1 and ID2
Input data ID-2 --> processing --> result comes from ID2 and ID1
Input data ID-3 --> processing --> result comes from ID3

### Steps to reproduce the problem

Just run the script again, but the IDs that are mixed up change each time. 

### Environment 

* Nextflow version: [19.10.0.5169 and 20.10.0.54.31] 
* Java version: [openjdk 11.0.10]
* Operating system: [Ubuntu]
* Bash version: (GNU bash version 4.4.20(1))

**Comment by pditommaso** [comment]
It's not clear the problem you are reporting. Please include a test script to replicate it.

**Comment by DanielAmsel** [comment, marked as answer]
Dear @pditommaso ,
I checked my files again and found that I obviously made the mistake that I simply added input channels to a process instead of combining them.
So nextflow took the first file that finished the previous process, I assume.

The issue can be closed, but thank you very much :)

Kind regards,
Daniel

**Reply by pditommaso** [reply]
Exactly, since the execution is parallel is output sequence is not deterministic. Have a look at the blue tip box at this link https://www.nextflow.io/docs/latest/process.html#input-of-generic-values
